packages/next/src/build/next-config-ts/utils.ts

```diff
@@ -0,0 +1,71 @@
+import type { Options as SWCOptions } from '@swc/core'
+import type { CompilerOptions } from 'typescript'
+
+import { resolve } from 'node:path'
+import { parseJsonFile } from '../load-jsconfig'
+
+export function resolveSWCOptions(cwd: string): SWCOptions {
+  const tsConfig = lazilyGetTSConfig(cwd)
+
+  return {
+    jsc: {
+      parser: {
+        syntax: 'typescript',
+      },
+      paths: tsConfig?.compilerOptions?.paths,
+      // SWC requires `baseUrl` to be passed when `paths` are used.
+      // Also, `baseUrl` must be absolute.
+      baseUrl: resolve(cwd, tsConfig?.compilerOptions?.baseUrl ?? ''),
+      experimental: {
+        keepImportAttributes: true,
```

Comment:
should we keep `keepImportAttributes` for next.config? it's not going to be bundled by bundler later. IMO they should be stripped

Reply:
It's needed for the loader to resolve to import JSON with an assertion:

```ts
import fooJson from './foo.json' with { type: 'json' }
```

Received Error without `keepImportAttributes`:

```shell
> Build error occurred
TypeError: Module "file:///.../foo.json" needs an import attribute of type "json"
    at load (node_modules/.pnpm/file+..+.../next/dist/build/next-config-ts/loader.mjs:56:16) {
  code: 'ERR_IMPORT_ASSERTION_TYPE_MISSING'
}
```

I added `emitAssertForImportAttributes` because as the `with` attribute was replaced from `assert` at v18.20.0, using `with` attribute received  an error:

```shell
> Build error occurred
[SyntaxError: Unexpected token 'with']
```

But, I guess the user can use the compatible assertion, so will remove `emitAssertForImportAttributes`.